Add SDK v5 integration steps

Sporiff opened the pull request:
Closes https://adjustcom.atlassian.net/browse/ADAPP-16715

This PR includes updates to:

- [x] The Android integration guide
- [x] The Android feature guides
- [x] The iOS integration guide
- [x] The iOS feature guides

Sporiff commented:
@shashanksu @uerceg Android instructions are done. The dev hub has a slightly different content setup, so here are the direct links to the stuff that was highlighted in your doc:

- [Integration guide](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android?version=v5)
- [COPPA compliance](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/features/privacy?version=v5#v5-coppa-compliance)
- [Play Store Kids app](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/features/privacy?version=v5#v5-play-store-kids-apps)
- [Revenue deduplication](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/features/events?version=v5#v5-deduplicate-revenue-events)
- [Reattribution using deep links](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/features/deep-links?version=v5#v5-reattribution-via-deep-links)
- [Global parameters](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/features/session-parameters?version=v5)
- [Offline mode](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/configuration?version=v5#v5-activate-offline-mode)
- [Device IDs](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/features/device-info?version=v5)
- [Disable SDK](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/configuration?version=v5#v5-disable-the-sdk). Enable step is just below.
- [Attribution information](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/features/attribution?version=v5)

shashanksu commented:
@Sporiff It looks good.  Just few of the minor observations though:
1. [Huawei Install Referrer](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android?version=v5#v5-huawei-referrer-api) and [Meta Install Referrer](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android?version=v5#v5-meta-referrer-integration) can also be consider as plugin, similar to how [Xiaomi IR plugin](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android?version=v5#v5-xiaomi-referrer-plugin) is placed in a plugin section.
2. [GetAttribution](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/features/attribution?version=v5#get-current-attribution-information) method's signature is also changed to accept callback, as mentioned in the [migration guide](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/migration/v4-to-v5#device-id-getters).
3. The example mentioned in the [LinkResolution](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/features/deep-links?version=v5#v5-link-resolution) also changes where reference to deeplink API is now changed from `Adjust.appWillOpenUrl` to `Adjust.processDeeplink`.
4. [Example](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/features/events?version=v5#v5-example-2) for setting Deduplication Id to be updated, replacing mention of `setOrderId` to `setDeduplicationId`.
5. [DisableThirdPartySharing](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/features/privacy?version=v5#v5-disable-third-party-sharing) do not exist in v5 as mentioned in the [migration guide](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/migration/v4-to-v5#disable-sharing-globally). 
6. [SetUrlStrategy](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/features/privacy?version=v5#v5-set-url-strategy) API is also changed as mentioned in the [migration guide](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/migration/v4-to-v5#url-strategy).
7. Reset method name for [Global Callback Parameter](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/features/session-parameters?version=v5#v5-reset-global-callback-parameters) and [Global Partner Parameter](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/features/session-parameters?version=v5#v5-reset-global-partner-parameters) is changed from `reset...` to `remove ... `  as mentioned in the [migration guide](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/migration/v4-to-v5#callback-params).

Sporiff commented:
@Aditi3 iOS has been updated in the same way. Specific links are:

- [Integration guide](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/ios?version=v5)
- [COPPA compliance](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/ios/features/privacy?version=v5#v5-coppa-compliance)
- [Revenue deduplication](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/ios/features/events?version=v5#v5-deduplicate-revenue-events)
- [Reattribution using deep links](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/ios/features/deep-links/direct?version=v5#v5-modify-your-ios-app)
- [Global parameters](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/ios/features/session-parameters?version=v5)
- [Offline mode](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/ios/configuration?version=v5#v5-activate-offline-mode)
- [Device IDs](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/ios/features/device-info?version=v5)
- [Disable SDK](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/ios/configuration?version=v5#v5-disable-the-sdk). Enable step is just below.
- [Attribution information](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/ios/features/attribution?version=v5)

Sporiff commented:
@shashanksu I've updated the preview to address what you raised here.

Aditi3 commented on src/content/docs/sdk/ios/index.mdx:
````diff
@@ -1,11 +1,348 @@
+```swift
+#import <AdjustSdk/Adjust.h>
+
+func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+   let yourAppToken = "{YourAppToken}"
+   let environment = ADJEnvironmentSandbox as? String
+   let adjustConfig = ADJConfig(
+         appToken: yourAppToken,
+         environment: environment)
+
+   Adjust.initSdk(adjustConfig)
+   // Your code here
+   return true
+}
````
```suggestion
func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
        // Override point for customization after application launch.
        let yourAppToken = "{YourAppToken}"
        let environment = ADJEnvironmentSandbox
        let adjustConfig = ADJConfig(appToken: yourAppToken,
                                     andEnvironment: environment)

        Adjust.initSdk(adjustConfig)
        // Your code here
        return true
    }
```

Aditi3 commented on src/content/docs/sdk/ios/index.mdx:
````diff
@@ -1,11 +1,348 @@
+```swift
+#import <AdjustSdk/Adjust.h>
+
+func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+   let yourAppToken = "{YourAppToken}"
+   let environment = ADJEnvironmentProduction as? String
+   let myAdjustConfig = ADJConfig(
+       appToken: yourAppToken,
+       environment: environment,
+       allowSuppressLogLevel: true)
+   myAdjustConfig?.logLevel = ADJLogLevelVerbose
+   //...
+   Adjust.initSdk(myAdjustConfig)
+   //...
+   return true
+}
````
```suggestion
func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
        // Override point for customization after application launch.
        let yourAppToken = "{YourAppToken}"
        let environment = ADJEnvironmentProduction
        let myAdjustConfig = ADJConfig(
            appToken: yourAppToken,
            environment: environment,
            andSuppressLogLevel: true)
        myAdjustConfig?.logLevel = ADJLogLevelVerbose
        //...
        Adjust.initSdk(myAdjustConfig)
        // Your code here
        return true
    }
```

Aditi3 commented on src/content/docs/sdk/ios/index.mdx:
````diff
@@ -1,11 +1,348 @@
+<Tab title="Swift" sync="swift">
+
+```swift
+func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+   let yourAppToken = "{YourAppToken}"
+
+   #if DEBUG
+   let environment = ADJEnvironmentSandbox as? String
+   let adjustConfig = ADJConfig(
+         appToken: yourAppToken,
+         environment: environment)
+   adjustConfig?.logLevel = ADJLogLevelVerbose
+
+   #else
+   let environment = ADJEnvironmentProduction as? String
+   let adjustConfig = ADJConfig(
+         appToken: yourAppToken,
+         environment: environment)
+   adjustConfig?.logLevel = ADJLogLevelSuppress
+   #endif
+   //...
+   Adjust.initSdk(adjustConfig)
+   //...
+   return true
+}
````
```suggestion
func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
        // Override point for customization after application launch.
        let yourAppToken = "{YourAppToken}"

        #if DEBUG
        let environment = ADJEnvironmentSandbox
        let adjustConfig = ADJConfig(
            appToken: yourAppToken,
            andEnvironment: environment)
        adjustConfig?.logLevel = ADJLogLevelVerbose

        #else
        let environment = ADJEnvironmentProduction
        let adjustConfig = ADJConfig(
            appToken: yourAppToken,
            andEnvironment: environment)
        adjustConfig?.logLevel = ADJLogLevelSuppress
        #endif
        //...
        Adjust.initSdk(adjustConfig)
        //...
        return true
}
```

Aditi3 commented on src/content/docs/sdk/ios/index.mdx:
```diff
@@ -1,11 +1,348 @@
+   [adjustConfig setLogLevel:ADJLogLevelSuppress];
+   #endif
+   //...
+   [Adjust initSdk:myAdjustConfig];
```
```suggestion
   [Adjust initSdk: adjustConfig];
```

Aditi3 commented on src/content/docs/sdk/ios/configuration.mdx:
````diff
@@ -4,9 +4,598 @@ description: Follow the guides in this section to configure the Adjust SDK.
+<CodeBlock highlight="{range: 3-5}">
+
+```swift
+let yourAppToken = "{YourAppToken}"
+let environment = ADJEnvironmentSandbox as? String
+let adjustConfig = ADJConfig(
+   appToken: yourAppToken,
+   environment: environment)
````
```suggestion
let yourAppToken = "{YourAppToken}"
let environment = ADJEnvironmentSandbox
let adjustConfig = ADJConfig(
    appToken: yourAppToken,
    andEnvironment: environment)
```

Aditi3 commented on src/content/docs/sdk/ios/configuration.mdx:
````diff
@@ -4,9 +4,598 @@ description: Follow the guides in this section to configure the Adjust SDK.
+<CodeBlock highlight="{range: 7}">
+
+```swift
+let yourAppToken = "{YourAppToken}"
+let environment = ADJEnvironmentSandbox as? String
+let adjustConfig = ADJConfig(
+   appToken: yourAppToken,
+   environment: environment)
+// ...
+adjustConfig?.logLevel = ADJLogLevelVerbose
````
```suggestion
let yourAppToken = "{YourAppToken}"
let environment = ADJEnvironmentSandbox
let adjustConfig = ADJConfig(
    appToken: yourAppToken,
    andEnvironment: environment)
// ...
adjustConfig?.logLevel = ADJLogLevelVerbose
```

Aditi3 commented on src/content/docs/sdk/ios/configuration.mdx:
````diff
@@ -4,9 +4,598 @@ description: Follow the guides in this section to configure the Adjust SDK.
+<CodeBlock highlight="{range: 7}">
+
+```swift
+let yourAppToken = "{YourAppToken}"
+let environment = ADJEnvironmentSandbox as? String
+let adjustConfig = ADJConfig(
+   appToken: yourAppToken,
+   environment: environment)
+// ...
+adjustConfig?.externalDeviceId = "yourExternalDeviceId"
````
```suggestion
let yourAppToken = "{YourAppToken}"
let environment = ADJEnvironmentSandbox
let adjustConfig = ADJConfig(
   appToken: yourAppToken,
   andEnvironment: environment)
// ...
adjustConfig?.externalDeviceId = "yourExternalDeviceId"
```

Aditi3 commented on src/content/docs/sdk/ios/configuration.mdx:
````diff
@@ -4,9 +4,598 @@ description: Follow the guides in this section to configure the Adjust SDK.
+<CodeBlock highlight="{range: 6}">
+
+```swift
+let yourAppToken = "{YourAppToken}"
+let environment = ADJEnvironmentSandbox as? String
+let adjustConfig = ADJConfig(
+   appToken: yourAppToken,
+   environment: environment)
+adjustConfig?.defaultTracker = "{Token}"
+Adjust.appDidLaunch(adjustConfig)
````
```suggestion
let yourAppToken = "{YourAppToken}"
let environment = ADJEnvironmentSandbox
let adjustConfig = ADJConfig(
   appToken: yourAppToken,
   andEnvironment: environment)
adjustConfig?.defaultTracker = "{Token}"
Adjust.appDidLaunch(adjustConfig)
```

Aditi3 commented on src/content/docs/sdk/ios/configuration.mdx:
````diff
@@ -4,9 +4,598 @@ description: Follow the guides in this section to configure the Adjust SDK.
+<CodeBlock highlight="{range: 7}">
+
+```swift
+let yourAppToken = "{YourAppToken}"
+let environment = ADJEnvironmentSandbox as? String
+let adjustConfig = ADJConfig(
+   appToken: yourAppToken,
+   environment: environment)
+// ...
+adjustConfig.enableCostDataInAttribution()
````
```suggestion
let yourAppToken = "{YourAppToken}"
let environment = ADJEnvironmentSandbox
let adjustConfig = ADJConfig(
   appToken: yourAppToken,
   andEnvironment: environment)
// ...
adjustConfig.enableCostDataInAttribution()
```

Aditi3 commented on src/content/docs/sdk/ios/configuration.mdx:
````diff
@@ -4,9 +4,598 @@ description: Follow the guides in this section to configure the Adjust SDK.
+<CodeBlock title="Method signature">
+
+```objc
++ (nullable ADJConfig *)configWithAppToken:(nonnull NSString *)appToken
+                               environment:(nonnull NSString *)environment
+                     allowSuppressLogLevel:(BOOL)allowSuppressLogLevel;
````
```suggestion
- (nullable ADJConfig *)initWithAppToken:(nonnull NSString *)appToken
                             environment:(nonnull NSString *)environment
                     andSuppressLogLevel:(BOOL)allowSuppressLogLevel;
```

Aditi3 commented on src/content/docs/sdk/ios/configuration.mdx:
````diff
@@ -4,9 +4,598 @@ description: Follow the guides in this section to configure the Adjust SDK.
+
+```objective-c
+NSString *yourAppToken = @"{YourAppToken}";
+NSString *environment = ADJEnvironmentSandbox;
+*adjustConfig = [ADJConfig configWithAppToken:yourAppToken
+                                  environment:environment];
````
```suggestion
 ADJConfig *adjustConfig = [[ADJConfig alloc] initWithAppToken:appToken
                                                   andEnvironment:environment];
```

Aditi3 commented on src/content/docs/sdk/ios/configuration.mdx:
````diff
@@ -4,9 +4,598 @@ description: Follow the guides in this section to configure the Adjust SDK.
+```objective-c
+NSString *yourAppToken = @"{YourAppToken}";
+NSString *environment = ADJEnvironmentSandbox;
+*adjustConfig = [ADJConfig configWithAppToken:yourAppToken
+                                 environment:environment];
````
```suggestion
 ADJConfig *adjustConfig = [[ADJConfig alloc] initWithAppToken:appToken
                                                   andEnvironment:environment];
```

Aditi3 commented on src/content/docs/sdk/ios/configuration.mdx:
````diff
@@ -4,9 +4,598 @@ description: Follow the guides in this section to configure the Adjust SDK.
+```objc
+NSString *yourAppToken = @"{YourAppToken}";
+NSString *environment = ADJEnvironmentSandbox;
+*adjustConfig = [ADJConfig configWithAppToken:yourAppToken
+                                 environment:environment];
````
```suggestion
 ADJConfig *adjustConfig = [[ADJConfig alloc] initWithAppToken:appToken
                                                   andEnvironment:environment];
```

Aditi3 commented on src/content/docs/sdk/ios/configuration.mdx:
````diff
@@ -4,9 +4,598 @@ description: Follow the guides in this section to configure the Adjust SDK.
+<CodeBlock highlight="{range: 3}">
+
+```objective-c
+*adjustConfig = [ADJConfig configWithAppToken:@"{YourAppToken}"
+                                 environment:ADJEnvironmentSandbox];
````
```suggestion
ADJConfig *adjustConfig = [[ADJConfig alloc] initWithAppToken:appToken
                                                   andEnvironment:environment];
```

Aditi3 commented on src/content/docs/sdk/ios/configuration.mdx:
````diff
@@ -4,9 +4,598 @@ description: Follow the guides in this section to configure the Adjust SDK.
+```objc
+NSString *yourAppToken = @"{YourAppToken}";
+NSString *environment = ADJEnvironmentSandbox;
+*adjustConfig = [ADJConfig configWithAppToken:yourAppToken
````
```suggestion
 ADJConfig *adjustConfig = [[ADJConfig alloc] initWithAppToken:appToken
                                                   andEnvironment:environment];
```

Aditi3 commented on src/content/docs/sdk/ios/features/att.mdx:
````diff
@@ -2,14 +2,204 @@
+<Tab title="Swift" sync="swift">
+
+```swift {7}
+let yourAppToken = "{YourAppToken}"
+let environment = ADJEnvironmentSandbox as? String
+let adjustConfig = ADJConfig(
+    appToken: yourAppToken,
+    environment: environment)
````
```suggestion
let yourAppToken = "{YourAppToken}"
let environment = ADJEnvironmentSandbox
let adjustConfig = ADJConfig(
    appToken: yourAppToken,
    andEnvironment: environment)
```

Aditi3 commented on src/content/docs/sdk/ios/features/att.mdx:
```diff
@@ -2,14 +2,204 @@
+//...
+adjustConfig?.attConsentWaitingInterval = 30
+//...
+Adjust.appDidLaunch(adjustConfig)
```
```suggestion
Adjust.initSdk(adjustConfig)```

Aditi3 commented on src/content/docs/sdk/ios/features/att.mdx:
````diff
@@ -2,14 +2,204 @@
+<Tab title="Objective-C" sync="objc">
+
+```objc {4}
+*adjustConfig = [ADJConfig configWithAppToken:@"{YourAppToken}"
+                                  environment:ADJEnvironmentSandbox];
+//...
+[adjustConfig setAttConsentWaitingInterval:30];
+//...
+[Adjust appDidLaunch:adjustConfig];
````
```suggestion
 ADJConfig *adjustConfig = [[ADJConfig alloc] initWithAppToken:@"{YourAppToken}"
                                                   andEnvironment:environment];
//...
[adjustConfig setAttConsentWaitingInterval:30];
//...
[Adjust initSdk:adjustConfig];
```

Aditi3 commented:
Hi @Sporiff 

The code snippet below needs to be updated across the documentation guide under Objective-C label -

from - 

```
*adjustConfig = [ADJConfig configWithAppToken:@"{YourAppToken}"
                                  environment:ADJEnvironmentSandbox];
```

to 

```
ADJConfig *adjustConfig = [[ADJConfig alloc] initWithAppToken:@"{YourAppToken}"
                                                   andEnvironment:ADJEnvironmentSandbox];
```

Aditi3 commented:
On Swift's side - 

instead of below -

```
let yourAppToken = "{YourAppToken}"
let environment = ADJEnvironmentSandbox as? String
let adjustConfig = ADJConfig(
   appToken: yourAppToken,
   environment: environment)
Adjust.appDidLaunch(adjustConfig)
```

should be - 

```
let yourAppToken = "{YourAppToken}"
let environment = ADJEnvironmentSandbox
let adjustConfig = ADJConfig(
    appToken: yourAppToken,
    andEnvironment: environment)
Adjust.initSdk(adjustConfig)
```

Aditi3 commented on src/content/docs/sdk/ios/features/events.mdx:
````diff
@@ -2,8 +2,812 @@
+<Tab title="Objective-C" sync="objc">
+
+```objc
+ADJEvent *event = [ADJEvent eventWithEventToken:@"abc123"];
+[Adjust trackEvent:event];
````
```suggestion
ADJEvent *event = [[ADJEvent alloc] initWithEventToken:@"abc123"];
[Adjust trackEvent:event];
```

Aditi3 commented on src/content/docs/sdk/ios/features/events.mdx:
````diff
@@ -2,8 +2,812 @@
+<Tab title="Objective-C" sync="objc">
+
+```objc
+ADJEvent *event = [ADJEvent eventWithEventToken:@"abc123"];
+[Adjust trackEvent:event];
````
```suggestion
ADJEvent *event = [[ADJEvent alloc] initWithEventToken:@"abc123"];
[Adjust trackEvent:event];
```

Aditi3 commented on src/content/docs/sdk/ios/features/events.mdx:
````diff
@@ -2,8 +2,812 @@
+<Tab title="Swift" sync="swift">
+
+```swift
+import Adjust
````
```suggestion
import AdjustSdk
```

Aditi3 commented on src/content/docs/sdk/ios/features/events.mdx:
````diff
@@ -2,8 +2,812 @@
+<Tab title="Objective-C" sync="objc">
+
+```objc
+ADJEvent *event = [ADJEvent eventWithEventToken:@"abc123"];
````
```suggestion
ADJEvent *event = [[ADJEvent alloc] initWithEventToken:@"abc123"];
```

Aditi3 commented on src/content/docs/sdk/ios/features/events.mdx:
````diff
@@ -2,8 +2,812 @@
+<Tab title="Swift" sync="swift">
+
+```swift
+import Adjust
````
```suggestion
import AdjustSdk
```

Aditi3 commented on src/content/docs/sdk/ios/features/events.mdx:
```diff
@@ -2,8 +2,812 @@
+@implementation ViewControllerObjC
+
+- (IBAction)clickRecordRevenueEvent:(UIButton *)sender {
+   ADJEvent *event = [ADJEvent eventWithEventToken:@"g3mfiw"];
```
```suggestion
    ADJEvent *event = [[ADJEvent alloc] initWithEventToken:@"g3mfiw"];
```

Aditi3 commented on src/content/docs/sdk/ios/features/events.mdx:
```diff
@@ -2,8 +2,812 @@
+@implementation ViewControllerObjC
+
+- (IBAction)clickRecordSimpleEvent:(UIButton *)sender {
+    ADJEvent *event = [ADJEvent eventWithEventToken: @"g3mfiw"];
```
```suggestion
    ADJEvent *event = [[ADJEvent alloc] initWithEventToken:@"g3mfiw"];
```

Aditi3 commented on src/content/docs/sdk/ios/features/events.mdx:
````diff
@@ -2,8 +2,812 @@
+<Tab title="Objective-C" sync="objc">
+
+```objc
+ADJEvent *event = [ADJEvent eventWithEventToken:@"abc123"];
````
```suggestion
ADJEvent *event = [[ADJEvent alloc] initWithEventToken:@"g3mfiw"];
```

Aditi3 commented on src/content/docs/sdk/ios/features/events.mdx:
```diff
@@ -2,8 +2,812 @@
+@implementation ViewControllerObjC
+
+- (IBAction)clickRecordCallbackEvent:(UIButton *)sender {
+   ADJEvent *event = [ADJEvent eventWithEventToken:@"g3mfiw"];
```
```suggestion
   ADJEvent *event = [[ADJEvent alloc] initWithEventToken:@"g3mfiw"];
```

Aditi3 commented on src/content/docs/sdk/ios/features/events.mdx:
````diff
@@ -2,8 +2,812 @@
+<Tab title="Objective-C" sync="objc">
+
+```objc
+ADJEvent *event = [ADJEvent eventWithEventToken:@"abc123"];
````
```suggestion
ADJEvent *event = [[ADJEvent alloc] initWithEventToken:@"abc123"];
```

Aditi3 commented on src/content/docs/sdk/ios/features/events.mdx:
```diff
@@ -2,8 +2,812 @@
+@implementation ViewControllerObjC
+
+- (IBAction)clickRecordPartnerParamsEvent:(UIButton *)sender {
+   ADJEvent *event = [ADJEvent eventWithEventToken:@"g3mfiw"];
```
```suggestion
   ADJEvent *event = [[ADJEvent alloc] initWithEventToken:@"g3mfiw"];
```

Aditi3 commented on src/content/docs/sdk/ios/features/events.mdx:
````diff
@@ -2,8 +2,812 @@
+<Tab title="Objective-C" sync="objc">
+
+```objc
+ADJEvent *event = [ADJEvent eventWithEventToken:@"abc123"];
````
```suggestion
ADJEvent *event = [[ADJEvent alloc] initWithEventToken:@"abc123"];
```

Aditi3 commented on src/content/docs/sdk/ios/features/events.mdx:
```diff
@@ -2,8 +2,812 @@
+@implementation ViewControllerObjC
+
+- (IBAction)clickRecordCallbackIdEvent:(UIButton *)sender {
+   ADJEvent *event = [ADJEvent eventWithEventToken:@"g3mfiw"];
```
```suggestion
   ADJEvent *event = [[ADJEvent alloc] initWithEventToken:@"g3mfiw"];
```

Aditi3 commented on src/content/docs/sdk/ios/features/privacy.mdx:
````diff
@@ -2,8 +2,605 @@
+<Tab title="Swift" sync="swift">
+
+```swift
+let adjustThirdPartySharing = ADJThirdPartySharing.initWithIsEnabledNumberBool(1)
+Adjust.trackThirdPartySharing(adjustThirdPartySharing)
````
```suggestion
let adjustThirdPartySharing = ADJThirdPartySharing(isEnabled: 1)
Adjust.trackThirdPartySharing(adjustThirdPartySharing)
```

Aditi3 commented on src/content/docs/sdk/ios/features/privacy.mdx:
````diff
@@ -2,8 +2,605 @@
+<Tab title="Objective-C" sync="objc">
+
+```objc
+ADJThirdPartySharing *adjustThirdPartySharing = [[ADJThirdPartySharing alloc] initWithIsEnabledNumberBool:1];
+[Adjust trackThirdPartySharing:adjustThirdPartySharing];
````
```suggestion
ADJThirdPartySharing *adjustThirdPartySharing = [[ADJThirdPartySharing alloc] initWithIsEnabled:@1];
[Adjust trackThirdPartySharing:adjustThirdPartySharing];
```

Aditi3 commented on src/content/docs/sdk/ios/features/privacy.mdx:
````diff
@@ -2,8 +2,605 @@
+<Tab title="Swift" sync="swift">
+
+```swift
+let adjustThirdPartySharing = ADJThirdPartySharing.initWithIsEnabledNumberBool(0)
+Adjust.trackThirdPartySharing(adjustThirdPartySharing)
````
```suggestion
let adjustThirdPartySharing = ADJThirdPartySharing(isEnabled: 0)
Adjust.trackThirdPartySharing(adjustThirdPartySharing)
```

Aditi3 commented on src/content/docs/sdk/ios/features/privacy.mdx:
````diff
@@ -2,8 +2,605 @@
+<Tab title="Objective-C" sync="objc">
+
+```objc
+ADJThirdPartySharing *adjustThirdPartySharing = [[ADJThirdPartySharing alloc] initWithIsEnabledNumberBool:0];
````
```suggestion
ADJThirdPartySharing *adjustThirdPartySharing = [[ADJThirdPartySharing alloc] initWithIsEnabled:@0];
```

Aditi3 commented on src/content/docs/sdk/ios/features/privacy.mdx:
````diff
@@ -2,8 +2,605 @@
+<Tab title="Swift" sync="swift">
+
+```swift
+let adjustThirdPartySharing = ADJThirdPartySharing.initWithIsEnabledNumberBool(nil)
````
```suggestion
let adjustThirdPartySharing = ADJThirdPartySharing(isEnabled:nil)
```

Aditi3 commented on src/content/docs/sdk/ios/features/privacy.mdx:
````diff
@@ -2,8 +2,605 @@
+<Tab title="Objective-C" sync="objc">
+
+```objc
+ADJThirdPartySharing *adjustThirdPartySharing = [[ADJThirdPartySharing alloc] initWithIsEnabledNumberBool:nil];
````
```suggestion
ADJThirdPartySharing *adjustThirdPartySharing = [[ADJThirdPartySharing alloc] initWithIsEnabled:nil];
```

Aditi3 commented on src/content/docs/sdk/ios/features/privacy.mdx:
````diff
@@ -2,8 +2,605 @@
+<Tab title="Objective-C" sync="objc">
+
+```objc
+ADJThirdPartySharing *tps = [[ADJThirdPartySharing alloc] initWithIsEnabledNumberBool:@YES];
````
```suggestion
ADJThirdPartySharing *tps = [[ADJThirdPartySharing alloc] initWithIsEnabled:@1];
```

Aditi3 commented on src/content/docs/sdk/ios/features/privacy.mdx:
````diff
@@ -2,8 +2,605 @@
+<Tab title="Swift" sync="swift">
+
+```swift
+let adjustThirdPartySharing = ADJThirdPartySharing.initWithIsEnabledNumberBool(nil)
````
```suggestion
let adjustThirdPartySharing = ADJThirdPartySharing(isEnabled:nil)
```

Sporiff commented:
@Aditi3 I've updated the `ADJConfig` examples as you've written them. I've also updated the Ad revenue guides for both iOS and Android @shashanksu

shashanksu commented:
@Sporiff Few more observations on Android docs, please find below:
1. Configuration - [Enable Cost Data Sending ](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/configuration?version=v5#v5-enable-cost-data-sending) `AdjustConfig.setNeedsCost(true)` is changed to `AdjustConfig.enableCostDataInAttribution()`.  Also, this can be added in the MigrationGuide.
2. Configuration - [Enable background recording](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/configuration?version=v5#v5-enable-background-recording) `AdjustConfig.setSendInBackground(true)` is changed to `AdjustConfig.enableSendingInBackground()`.  Also, this can be added in the MigrationGuide.
3. Configuration - [Enable event buffering](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/configuration?version=v5#v5-enable-event-buffering)  This is removed in v5 (as already mentioned in the Migration Guide).
4. Configuration - [Delay start of the SDK](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/configuration?version=v5#v5-delay-the-start-of-the-sdk) This is removed in v5 (as already mentioned in the Migration Guide).
5. [Multi Process Apps](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/setup/multi-process) All instances of `Adjust.onCreate(config)` to be replaced with `Adjust.initSdk(config)`
6. [Send preinstalled app activity](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/setup/preinstalled) All instances of `Adjust.onCreate(config)` to be replaced with `Adjust.initSdk(config)`
7. [Get attribution information](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/features/attribution?version=v5) All instances of `Adjust.onCreate(config)` to be replaced with `Adjust.initSdk(config)`
8. [Send callback information](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/features/callbacks) All instances of `Adjust.onCreate(config)` to be replaced with `Adjust.initSdk(config)`
9. [Admob SDK integration](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/integrations/admob)  `AdjustConfig.AD_REVENUE_ADMOB` usage in the example to be replaced with `"admob_sdk"`
10. [AD(X) SDK integration](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/integrations/adx) `AdjustConfig.AD_REVENUE_ADX` usage to be replaced with `"adx_sdk"`
11. [AppLovin MAX SDK integration](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/integrations/applovin) `AdjustConfig.AD_REVENUE_APPLOVIN_MAX` usage to be replaced with `"applovin_max_sdk"`
12. [Batch.com SDK integration](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/integrations/batch) All instances of `Adjust.onCreate(config)` to be replaced with `Adjust.initSdk(config)`
13. [Helium by Chartboost SDK integration](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/integrations/helium) `AdjustConfig.AD_REVENUE_HELIUM_CHARTBOOST` usage to be replaced with `"helium_chartboost_sdk"`
14. [Ironsource SDK integration ](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/integrations/ironsource) `AdjustConfig.AD_REVENUE_IRONSOURCE ` usage to be replaced with `"ironsource_sdk"`
15. [Mixpanel SDK integration](https://dev-docs-git-adapp-16715-sdk-v-5-integra-00eac7-product-content.vercel.app/en/sdk/android/integrations/mixpanel) `AdjustConfig.AD_REVENUE_IRONSOURCE ` usage to be replaced with `"ironsource_sdk"`

Aditi3 commented on src/content/docs/sdk/ios/features/preinstalled.mdx:
```diff
@@ -2,14 +2,65 @@
+ADJConfig *adjustConfig = [[ADJConfig alloc] initWithAppToken:@"{YourAppToken}"
+                                                   andEnvironment:ADJEnvironmentSandbox];
+[adjustConfig setDefaultTracker:@"{Token}"];
+[Adjust appDidLaunch:adjustConfig];
```
```suggestion
[Adjust initSdk:adjustConfig];
```

Aditi3 commented on src/content/docs/sdk/ios/features/skad.mdx:
```diff
@@ -2,8 +2,190 @@
+//...
+adjustConfig.deactivateSKAdNetworkHandling()
+//...
+Adjust.appDidLaunch(adjustConfig)
```
```suggestion
Adjust.initSdk(adjustConfig)
```

Aditi3 commented on src/content/docs/sdk/ios/features/skad.mdx:
```diff
@@ -2,8 +2,190 @@
+                                                   andEnvironment:ADJEnvironmentSandbox];
+[adjustConfig.deactivateSKAdNetworkHandling];
+/// ...
+[Adjust appDidLaunch:adjustConfig];
```
```suggestion
[Adjust initSdk:adjustConfig];
```

Aditi3 commented on src/content/docs/sdk/ios/index.mdx:
```diff
@@ -1,11 +1,350 @@
+
+1. Declare your config object in the `didFinishLaunching` or `didFinishLaunchingWithOptions` method of your app delegate.
+2. Set the `logLevel` property on your config object to `ADJLogLevelVerbose` (verbose). You must enable verbose logging to retrieve device information.
+3. Pass the config object as an argument to the `appDidLaunch` method.
```
```suggestion
3. Pass the config object as an argument to the `initSdk` method.
```

Aditi3 commented on src/content/docs/sdk/ios/configuration.mdx:
```diff
@@ -4,9 +4,598 @@ description: Follow the guides in this section to configure the Adjust SDK.
+   appToken: yourAppToken,
+   andEnvironment: environment)
+adjustConfig?.defaultTracker = "{Token}"
+Adjust.appDidLaunch(adjustConfig)
```
```suggestion
Adjust.initSdk(adjustConfig)
```

Aditi3 commented on src/content/docs/sdk/ios/configuration.mdx:
```diff
@@ -4,9 +4,598 @@ description: Follow the guides in this section to configure the Adjust SDK.
+ADJConfig *adjustConfig = [[ADJConfig alloc] initWithAppToken:appToken
+                                                   andEnvironment:environment];
+[adjustConfig setDefaultTracker:@"{Token}"];
+[Adjust appDidLaunch:adjustConfig];
```
```suggestion
[Adjust initSdk:adjustConfig];
```

Aditi3 commented on src/content/docs/sdk/ios/features/skad.mdx:
````diff
@@ -2,8 +2,190 @@
+<CodeBlock title="Method signature">
+
+```objc
+- (void)deactivateSKAdNetworkHandling;
````
```suggestion
- (void)disableSkanAttribution;
```

Aditi3 commented on src/content/docs/sdk/ios/features/skad.mdx:
```diff
@@ -2,8 +2,190 @@
+
+<Callout type="important">
+
+You must call the `deactivateSKAdNetworkHandling` method _before_ initializing the Adjust SDK.
```
```suggestion
You must call the `disableSkanAttribution` method _before_ initializing the Adjust SDK.
```

Aditi3 commented on src/content/docs/sdk/ios/features/skad.mdx:
```diff
@@ -2,8 +2,190 @@
+    appToken: yourAppToken,
+    andEnvironment: environment)
+//...
+adjustConfig.deactivateSKAdNetworkHandling()
```
```suggestion
adjustConfig.disableSkanAttribution()
```

Aditi3 commented on src/content/docs/sdk/ios/features/skad.mdx:
````diff
@@ -2,8 +2,190 @@
+```objective-c
+ADJConfig *adjustConfig = [[ADJConfig alloc] initWithAppToken:@"{YourAppToken}"
+                                                   andEnvironment:ADJEnvironmentSandbox];
+[adjustConfig.deactivateSKAdNetworkHandling];
````
```suggestion
[adjustConfig disableSkanAttribution];
```

Aditi3 commented on src/content/docs/sdk/ios/features/skad.mdx:
```diff
@@ -2,8 +2,190 @@
+
+The Adjust SDK communicates with SKAdNetwork by default. The SDK registers for SKAdNetwork attribution upon initialization.
+
+Your config object contains a boolean `isSKAdNetworkHandlingActive` property that controls this behavior. You can disable SKAdNetwork communication by calling the `deactivateSKAdNetworkHandling` method with no argument.
```
```suggestion
You can disable SKAdNetwork communication by calling the `disableSkanAttribution ` method with no argument.
```

Sporiff commented:
@shashanksu I've updated everything you've mentioned there, plus I've been through and cleaned up all other `onCreate` calls and integration docs.

@Aditi3 Thanks for the changes. I've updated the URL strategy and integration docs.

Aditi3 reviewed (changes requested):
Thank you for implementing all the changes. This is the last review pointer. 

Change the below everywhere - 

Adjust.appDidLaunch(adjustConfig) to Adjust.initSdk(adjustConfig)

Aditi3 reviewed (approved):
Looks good!🚀
